### ISSUE TITLE
ERROR:  Could not find a valid gem 'engineyard-serverside-adapter' (= 1.4.10) in any repository

### ISSUE THREAD

**Issue by jaustinhughey** [engineyard/engineyard#192]
```
jaustinhughey@prometheus:~/Projects/foo$ gem install engineyard
ERROR:  Could not find a valid gem 'engineyard-serverside-adapter' (= 1.4.10) in any repository
ERROR:  Possible alternatives: engineyard-serverside-adapter
```

This is on a brand new install of Ubuntu Linux. Environment info:

```
jaustinhughey@prometheus:~/Projects/foo$ which ruby
/home/jaustinhughey/.rbenv/shims/ruby
jaustinhughey@prometheus:~/Projects/foo$ ruby -v
ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-linux]
jaustinhughey@prometheus:~/Projects/foo$ gem -v
2.4.5
```

**Comment by martinemde** [comment]
That's a very old version, and it seems to be required by a very old yanked version of engineyard gem 1.3.26.

Here's the weird thing about it all:

```
~: gem list -r engineyard

*** REMOTE GEMS ***

engineyard (3.1.2, 1.3.26)
```

It's only showing the most recent version *and* the yanked version with this problem (the engineyard gem version is also yanked, the one matching with the yanked ey-ss-adapter). To confound this issue further, this version 1.3.26 is even succeeded by a number of 1.3.x gems that were not yanked:

> 1.3.33 - September 27, 2011 (43.5 KB)
> 1.3.32 - September 23, 2011 (43.5 KB)
> 1.3.31 - September 21, 2011 (43 KB)
> 1.3.30 - September 19, 2011 (43 KB)
> 1.3.29 - September 15, 2011 (43 KB)
> 1.3.28 - September 14, 2011 (43 KB)
> 1.3.27 - September 14, 2011 (43 KB) yanked
> 1.3.26 - September 13, 2011 (43 KB) yanked
> 1.3.25 - September 12, 2011 (44 KB)

I even tried yanking these older versions again, to no avail.

```
~: gem yank engineyard -v 1.3.26
Yanking gem from https://rubygems.org...
The version 1.3.26 has already been yanked.
```

**Comment by martinemde** [comment]
Rubygems issue: http://help.rubygems.org/discussions/problems/20513-rubygems-automatically-selects-a-4-year-old-yanked-version-of-our-gem-for-the-default-install?unresolve=true

**Comment by evanphx** [comment]
I'm not able to replicate the issue with rubygems today:

```
zero :: ~/tmp> gem install -i gems --no-doc engineyard 
Fetching: highline-1.6.21.gem (100%)
Successfully installed highline-1.6.21
Fetching: escape-0.0.4.gem (100%)
Successfully installed escape-0.0.4
Fetching: multi_json-1.11.0.gem (100%)
Successfully installed multi_json-1.11.0
Fetching: engineyard-serverside-adapter-2.4.0.gem (100%)
Successfully installed engineyard-serverside-adapter-2.4.0
Fetching: netrc-0.10.3.gem (100%)
Successfully installed netrc-0.10.3
Fetching: mime-types-2.4.3.gem (100%)
Successfully installed mime-types-2.4.3
Fetching: rest-client-1.7.3.gem (100%)
Successfully installed rest-client-1.7.3
Fetching: engineyard-cloud-client-2.1.1.gem (100%)
Successfully installed engineyard-cloud-client-2.1.1
Fetching: net-ssh-2.9.3.beta1.gem (100%)
Successfully installed net-ssh-2.9.3.beta1
Fetching: addressable-2.3.7.gem (100%)
Successfully installed addressable-2.3.7
Fetching: launchy-2.4.3.gem (100%)
Successfully installed launchy-2.4.3
Fetching: engineyard-3.1.3.gem (100%)
Thanks for installing the engineyard gem!

Need help? Problem? https://support.cloud.engineyard.com
Not working? Uninstall this version to downgrade.

IMPORTANT:
  Please verify eydeploy.rb files in staging after upgrading!
  Internals frequently change that could break eydeploy.rb.

Getting started:
  1. Run `ey init` to generate an annotated config file.
  2. Review the ey.yml file and commit it to your repository.
  3. Run `ey deploy` when you're ready to deploy.
The Engine Yard Pandas want to help you!
Email pandas@engineyard.com with your questions.
Successfully installed engineyard-3.1.3
12 gems installed
```

Do you have the output from `gem install engineyard` when the issue happened? I'm wondering if maybe one of the dependent gems had a requirement that cause the resolver to only find 1.3.26 as a valid path, but that requirement has since been corrected (the requirement could be in any gem seen while attempt to calculate the gems to install, i.e. the transitive closure).

**Comment by martinemde** [comment]
@evanphx Thank you for responding. I think we just narrowed it down to our internal EY gem server, which purports to have engineyard 1.3.26 though it has already been deleted. It seems to be choosing this version over the versions on rubygems.org for default install. Is there a way to set priority in the .gemrc file? Order seems to not matter.

Thanks for your attention to this. I'll resolve the issue on our end. Closing this issue as solved.
